crates/ruff_server/docs/setup/NEOVIM.md

```diff
@@ -0,0 +1,52 @@
+## NeoVim Setup Guide for `ruff server`
```

Comment:
I think the heading levels should just start with 1 and so on. Can you update that along with other headers?

Reply:
Hmmm I changed this but on second though I actually prefer the smaller headers. Heading level 1 just looks too big for a document of this size. The smaller headers would also match the header size in the `README`.